qemu/tests/hpt_miscellaneous.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    QEMU HPT miscellaneous test:
+    Scenario 1
+    1) Start a HPT guest on Power8 or Power9 host
+    2) Increase HPT
+    3) Reboot
+
+    Scenario 2
+    1)Start a HPT guest on Power8 or Power9 host
+    4) Reduce HPT
+    5) Reboot
+
+    Scenario 3
+    6) Hotplug 1G memory to the guest
+    3) Increase HPT
+    4) Decrease HPT
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    try:
+        error_context.context("Start miscellaneous HPT test...", logging.info)
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        session = vm.wait_for_login(timeout=600)
+        get_cmd = "cat /sys/kernel/debug/powerpc/hpt_order"
+        set_cmd = "echo %s > /sys/kernel/debug/powerpc/hpt_order"
+        st, get_hpt_def = session.cmd_status_output(get_cmd, timeout=30)
+        if st:
+            test.cancel("Can't find available HPT value so skip test")
+        else:
+            if params.get("mistype") == "mem":
+                mem_sub_hpt(test, vm)
+            hpt_size = int(get_hpt_def)
+            logging.debug("Default hpt size : '%s'" % get_hpt_def)
+            increment = params.get("increment").split()
+            for increm in increment:
+                hpt_size = hpt_size + int(increm)
+                set_hpt(vm, hpt_size, set_cmd)
+                get_hpt(test, session, hpt_size, get_cmd)
+            if "reboot" in params.get("mistype"):
+                vm.reboot()
+                vm.verify_alive()
+                session = vm.wait_for_login(timeout=600)
+                get_hpt(test, session, int(get_hpt_def), get_cmd)
+            vm.verify_kernel_crash()
+    except Exception as detail:
+        test.fail("Failed in HPT's miscellaneous tests"
```

Comment:
I think this should be a `test.error()`,because most of exceptions could be raised would be python run-time errors, which are not related to the test case itself.
So I'd suggest you remove the whole `try ... except...` block.